app/Platform/Services/PlayerGameActivityService.php

```diff
@@ -61,7 +63,35 @@ public function initialize(User $user, Game $game, bool $withSubsets = false): v
             $gameIds[] = $game->id;
         }
 
-        $playerSessions = $user->playerSessions()->with('gameHash')->whereIn('game_id', $gameIds)->orderBy('created_at')->get();
+        // Get the most recent reset for this user and game(s).
+        // If the player has a recent reset for any of the games, we only
+        // want to include sessions from after the most recent reset.
+        // When `$withSubsets` is true, we need to check all gameIds.
+        $lastReset = null;
+        if ($withSubsets && count($gameIds) > 1) {
+            // If we're dealing with subsets, check for resets on any of the games.
+            $lastReset = PlayerProgressReset::where('user_id', $user->id)
+                ->where(function ($q) use ($gameIds) {
+                    $q->where(function ($subQuery) use ($gameIds) {
+                        $subQuery->where('type', PlayerProgressResetType::Game)
+                            ->whereIn('type_id', $gameIds);
+                    })
+                    ->orWhere('type', PlayerProgressResetType::Account);
+                })
+                ->orderByDesc('created_at')
+                ->first();
+        } else {
+            $lastReset = PlayerProgressReset::forUserAndGame($user, $game)->first();
+        }
```

Comment:
TODO: reset events should be captured in the timeline and shown on the game actiivity log for the user.

Reply:
Great idea - I've added something in my backlog for this.